src/main/java/com/uid2/shared/store/EncryptedScopedStoreReader.java

```diff
@@ -0,0 +1,88 @@
+package com.uid2.shared.store;
+
+import com.uid2.shared.cloud.DownloadCloudStorage;
+import com.uid2.shared.model.S3Key;
+import com.uid2.shared.store.parser.Parser;
+import com.uid2.shared.store.parser.ParsingResult;
+import com.uid2.shared.store.scope.StoreScope;
+import com.uid2.shared.store.reader.RotatingS3KeyProvider;
+import io.vertx.core.json.JsonObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+
+import com.uid2.shared.encryption.AesGcm;
+
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Map;
+
+public class EncryptedScopedStoreReader<T> extends ScopedStoreReader<T> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(EncryptedScopedStoreReader.class);
+
+    private final int siteId;
+    private final RotatingS3KeyProvider s3KeyProvider;
+
+    public EncryptedScopedStoreReader(DownloadCloudStorage fileStreamProvider, StoreScope scope, Parser<T> parser, String dataTypeName, int siteId, RotatingS3KeyProvider s3KeyProvider) {
+        super(fileStreamProvider, scope, parser, dataTypeName);
+        this.siteId = siteId;
+        this.s3KeyProvider = s3KeyProvider;
+    }
+
+    @Override
+    protected long loadContent(String path) throws Exception {
+        try (InputStream inputStream = this.contentStreamProvider.download(path)) {
```

Comment:
Please be consistent with the use of this. - some places you use it, others not. I prefer to use this., but don't mind if you dont, as long as you are consistent